core/include/detray/utils/tuple_helpers.hpp

```diff
@@ -60,16 +68,18 @@ template <int N, typename... value_types>
 struct tuple_element<N, std::tuple<value_types...>>
     : std::tuple_element<N, std::tuple<value_types...>> {};
 
-// thrust::tuple
+// detray::tuple
 template <int N, typename... value_types>
-struct tuple_element<N, thrust::tuple<value_types...>>
-    : thrust::tuple_element<N, thrust::tuple<value_types...>> {};
+struct tuple_element<N, tuple<value_types...>> {
+    using type = std::decay_t<decltype(
+        ::detray::get<N>(std::declval<tuple<value_types...>>()))>;
```

Comment:
This only works for default constructible elements, right? So, say we have a ```vecmem::vector``` type in the tuple that needs a memory resource, this would not work? I am not saying that we need it, just trying to understand

Reply:
This also works for non-constructible elements! `std::declval<T>()` produces something that behaves like a valid `T`, even if `T` cannot be constructed from nothing; see also [the example on cppreference](https://en.cppreference.com/w/cpp/utility/declval).